ElectronicObserver/Data/Battle/BattleRankPrediction.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ElectronicObserver.Window.Dialog.QuestTrackerManager.Enums;
+using ElectronicObserver.Window.Tools.SortieRecordViewer;
+using ElectronicObserverTypes;
+
+namespace ElectronicObserver.Data.Battle;
+
+public class BattleRankPrediction
+{
+	public required IFleetData FriendlyMainFleetBefore { get; init; }
+	public required IFleetData FriendlyMainFleetAfter { get; init; }
+
+	public IFleetData? FriendlyEscortFleetBefore { get; init; }
+	public IFleetData? FriendlyEscortFleetAfter { get; init; }
+
+	public required IFleetData EnemyMainFleetBefore { get; init; }
+	public required IFleetData EnemyMainFleetAfter { get; init; }
+
+	public IFleetData? EnemyEscortFleetBefore { get; init; }
+	public IFleetData? EnemyEscortFleetAfter { get; init; }
+	
+	public int FriendlyShipCount { get; private set; }
+	public int FriendlyShipSunk { get; private set; }
+	public int FriendlyHpBefore { get; private set; }
+	public int FriendlyHpAfter { get; private set; }
+	public double FriendHpRate => (double)(FriendlyHpBefore - FriendlyHpAfter) / FriendlyHpBefore;
+
+	public int EnemyShipCount { get; private set; }
+	public int EnemyShipSunk { get; private set; }
+	public int EnemyHpBefore { get; private set; }
+	public int EnemyHpAfter { get; private set; }
+	public double EnemyHpRate => (double)(EnemyHpBefore - EnemyHpAfter) / EnemyHpBefore;
+
+	public BattleRank PredictRank()
+	{
+		ResetValues();
+
+		CalculeFriendlyMainFleetHp();
+		CalculeFriendlyEscortFleetHp();
+
+		CalculateEnemyMainFleetHp();
+		CalculateEnemyEscortFleetHp();
+
+		return GetWinRank();
+	}
+
+	public BattleRank PredictRankAirRaid()
+	{
+		ResetValues();
+		CalculeFriendlyMainFleetHp();
+		CalculeFriendlyEscortFleetHp();
+
+		double friendrate = (double)(FriendlyHpBefore - FriendlyHpAfter) / FriendlyHpBefore;
+
+		return GetWinRankAirRaid(friendrate);
+	}
+	
+	private void CalculeFriendlyMainFleetHp()
```

Comment:
This and CalculeFriendlyEscortFleetHp seem to be exactly the same.
You could just make
```cs
CalculeFriendlyFleetHp(IFleetData fleetBefore, IFleetData fleetAfter)
```
and just pass the fleets.